async.go

```diff
@@ -0,0 +1,136 @@
+package plc
+
+import (
+	"golang.org/x/sync/errgroup"
+)
+
+type asyncer interface {
+	Wait() error
+	Add(name string, value interface{})
+	AddError(err error)
+}
+
+type action func(string, interface{}) error
+
+type job struct {
+	name  string
+	value interface{}
+}
+
+var noJob = job{}
+
+type async struct {
+	action
+	*errgroup.Group
+	jobs        chan job
+	numRoutines int
+}
+
+func newAsync(act action) async {
+	as := async{
+		action:      act,
+		Group:       &errgroup.Group{},
+		jobs:        make(chan job, 1),
+		numRoutines: 1,
+	}
+
+	as.launchRoutine(noJob)
+
+	go func() {
+		// Listen for new jobs and launch them in new goroutines until the limit has been reached.
+		for newJob := range as.jobs {
+			as.numRoutines++
+			as.launchRoutine(newJob)
+		}
+	}()
+
+	return as
+}
+
+func (as async) launchRoutine(newJob job) {
+	as.Group.Go(func() error {
+		if newJob != noJob {
+			// First handle the provided job.
+			if err := as.takeAction(newJob); err != nil {
+				return err
+			}
+		}
+
+		// Now keep listening for new jobs
+		for j := range as.jobs {
+			if err := as.takeAction(j); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+	var maxRoutines = 128
+	if as.numRoutines >= maxRoutines {
```

Comment:
This does not do what you think it does :P

This is outside of the group. This check probably makes more sense in the go function above.

Reply:
This code was originally in the go function, but during refactoring I accidentally moved too much into a separate function. (I'd add a test to protect against this in the future, but that doesn't seem straightforward...or even possible?)